ctapipe/io/eventsource.py

```diff
@@ -205,6 +209,22 @@ def subarray(self) -> SubarrayDescription:
 
         """
 
+    @property
+    @abstractmethod
+    def observation_block(self) -> Dict[int, ObservationBlockContainer]:
+        """
+        Obtain the ObservationConfigurations from the EventSource, indexed by obs_id
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def scheduling_block(self) -> Dict[int, SchedulingBlockContainer]:
```

Comment:
We have the plural in `obs_ids`.

Also, since we are now requiring `observation_blocks` we could remove the `abstractmethod` from `def obs_ids` and just `return self.observation_blocks.keys()`

Reply:
good point - I'll drop the requirement to fill obs_ids and make it a non-abstract property of the base class